sdk/python/feast/errors.py

```diff
@@ -67,3 +69,11 @@ def __init__(self, offline_store_name: str, data_source_name: str):
         super().__init__(
             f"Offline Store '{offline_store_name}' does not support data source '{data_source_name}'"
         )
+
+
+class FeastJoinKeysDuringMaterialization(Exception):
+    def __init__(self, join_key_columns: Set[str], source_columns: Set[str]):
+        super().__init__(
+            f"The DataFrame must have at least {join_key_columns} columns present, "
```

Comment:
`The DataFrame` seems a bit unclear. Is it possible for us to be more specific?

Reply:
Yeah `The Dataframe being materialized`?

Reply:
I mean what is `The DataFrame`?

Reply:
Yeah added that